src/components/contact-datalist-item/ContactDatalistItem.tsx

```diff
@@ -0,0 +1,99 @@
+import * as React from 'react';
+import Avatar from '../avatar';
+import DatalistItem from '../datalist-item';
+
+import { ContactType } from '../../features/unified-share-modal/types';
+import { SuggestedPillType } from '../pill-selector-dropdown/types';
+
+import './ContactDatalistItem.scss';
+
+interface ContactDatalistItemProps {
+    getContactAvatarUrl?: (contact: ContactType) => string | Promise<string | null | undefined>;
+    getPillImageUrl?: (data: SuggestedPillType) => string | Promise<string>;
+    id?: string;
+    isExternal?: boolean;
+    name: string | null | undefined;
+    showAvatar?: boolean;
+    subtitle?: React.ReactNode;
+}
+
+interface ContactDatalistItemState {
+    avatarUrl: string | null | undefined;
+}
+
+class ContactDatalistItem extends React.PureComponent<ContactDatalistItemProps, ContactDatalistItemState> {
+    constructor(props: ContactDatalistItemProps) {
+        super(props);
+        this.state = { avatarUrl: undefined };
+    }
+
+    isMounted = false;
```

Comment:
Do we need to specify a type boolean?

Reply:
Looking at other components, it looks like we don't really specify the type for properties of a class. But is that weird not to?

Reply:
I think there was a type there at one point but a ts error made me remove it because its assumed

Reply:
Gotcha, yeah it looks like other components omit the type for these properties so it should be fine here as well.